### DISCUSSION TITLE
How to get the modelspace extents

### DISCUSSION THREAD

**Discussion by filippopoggini** [mozman/ezdxf#813] [Q&A]
How we could calculate the size of the draw in a dxf file?
We would like to know the coordinates x and y for any shape:

- top left point
- top right point
- bottom left point
- bottom right point

In this case, we could calculate the max square area of the draw.
Thanks in advance for your help.

**Comment by AGeissler** [comment]
This is not really an issue, but a "how to" question. Have you looked in the documentation and searched for "extent"? There are several entries for this, possibly e.g. "Calculate Extents for the Modelspace" will meet your needs?

**Comment by filippopoggini** [comment]
I will look documentation about extent.
Thanks

**Comment by filippopoggini** [comment]
I tried to get limit of shape in my dxf with code below:

```
print(doc.header['$EXTMAX']) #output: (8.5, 11.0)
print(doc.header['$EXTMIN']) #output: (0.0, 0.0)
print(doc.header['$LIMMAX']) #output: KeyError: '$LIMMAX'
print(doc.header['$LIMMIN']) #output: KeyError: '$LIMMAX'
print(doc.header['$PEXTMAX']) #output: KeyError: '$LIMMAX'
print(doc.header['$PLIMMIN']) #output: KeyError: '$LIMMAX'
print(doc.header['$PLIMMAX']) #output: KeyError: '$LIMMAX'
```
 I suppose that some keys do not exist in my dxf file `(8.5, 11.0)`  and `(0.0, 0.0)` do not really are limits of my shape.

I attach my sample shape (a star), how could I calculate the width and height of my shape?

![star-limit](https://user-images.githubusercontent.com/110598707/209785135-46c933a6-a357-4e2c-b0a6-4af467b10cc6.png)

**Reply by mozman** [reply]
The values stored in the HEADER section are not reliable, many third party libraries do not maintain these values and neither does `ezdxf` automatically.

Use the [bbox](https://ezdxf.mozman.at/docs/bbox.html) module to get the extents of DXF entities. For calculating the bounding box of the modelspace use:

```Python
from ezdxf import bbox

# -x-x-x- snip

bounding_box = bbox.extents(doc.modelspace())
```

BTW: when you edit the output of your script, please be careful, if the errors are not as obvious as in this case, it might waste a lot of time of someone trying to help you (most likely me).

**Reply by filippopoggini** [reply]
Dear Manfred,
`bbox` was the solution, it works awesome.

Thanks for your support.

Best regards
